crates/nuts/src/nut00/secret.rs

```diff
@@ -145,6 +149,20 @@ impl AsRef<str> for Secret {
     }
 }
 
+// Secret implementation
+impl ToSql for Secret {
+    fn to_sql(&self) -> Result<ToSqlOutput<'_>> {
+        Ok(self.to_string().into())
+    }
+}
+
+impl FromSql for Secret {
+    fn column_result(value: ValueRef<'_>) -> FromSqlResult<Self> {
+        String::column_result(value)
+            .and_then(|s| Secret::new(&s).map_err(|_| FromSqlError::InvalidType))
```

Comment:
`FromSqlError::Other` is the error you should be using there
https://docs.rs/rusqlite/latest/rusqlite/types/enum.FromSqlError.html#variants